src/awkward/operations/str/__init__.py

```diff
@@ -73,6 +83,9 @@
 from awkward.operations.str.akstr_match_substring_regex import *
 from awkward.operations.str.akstr_starts_with import *
 
+# dictionary-encoding (exclusively for strings)
+from awkward.operations.str.akstr_to_categorical import *
```

Comment:
Good: we'll introduce the new function under `ak.str` and deprecate the old one. (No overlapping.)

Reply:
Yep, it's in motion.